### ISSUE TITLE
Sandstorm seems to hang

### ISSUE THREAD

**Issue by frilous** [sandstorm-io/sandstorm#3061]
Hello,

I just installed sandstorm.
The server is running ubuntu 16.04
Each time I launch the app (under firefox), I have a kind of wheel that keeps turning, but nothing else appens.
I had to reconfigure my server to accept wildcards. It does now (ping toto.server is ok), but it didn't solve the problem.
I don't have any message on the main screen, but on the configuration page, I still have the messge about the wildcards (bad configuration).

Would you hava any advice to solve the problem, please ?
Thanks
Frilous

**Comment by ocdtrekkie** [comment]
Can you share the configuration you're using?

**Comment by frilous** [comment]
Hello,

 The server is running Ubuntu 16.04.
Java is : 
openjdk version "1.8.0_162"
OpenJDK Runtime Environment (build 1.8.0_162-8u162-b12-0ubuntu0.16.04.2-b12)
OpenJDK 64-Bit Server VM (build 25.162-b12, mixed mode)

The client : 
windows 7 pro 64bits
Firefox 56
java version "1.8.0_151"
Java(TM) SE Runtime Environment (build 1.8.0_151-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.151-b12, mixed mode)

I tried on another computer running ubuntu, but it doesn't work either. 
Actualy, it works only on the server web-browser...
On the other clients, another thing that doesn't work, is that the icos doesn't appear in the Apps window...

Thank you for your help...
Frilous

**Comment by ocdtrekkie** [comment]
Sandstorm doesn't care about Java, as the case is. (And for the most part, it packs in all its own dependencies.)

I meant can you share your sandstorm.conf file? If you need to replace domain names with fake ones, you can do so. The docs on how the sandstorm.conf file works is here: https://docs.sandstorm.io/en/latest/administering/config-file/

**Comment by frilous** [comment]
oh, ok, here is the file : 
SERVER_USER=sandstorm
PORT=6080
MONGO_PORT=6081
BIND_IP=0.0.0.0
BASE_URL=http://emhcalcul:6080
WILDCARD_HOST=*.emhcalcul:6080
UPDATE_CHANNEL=dev
ALLOW_DEV_ACCOUNTS=false
SMTP_LISTEN_PORT=25

Thanks

**Comment by ocdtrekkie** [comment]
Hmm, that seems right to me, assuming you have nothing like Nginx or the like in front of it. I take it the Sandstorm UI (grains list page/etc.) works, but just launching apps hangs? I feel like there still has to be something wrong with DNS if it's giving you that warning.

**Comment by frilous** [comment]
Yes, That's what I think too, even though the ping anything.example.com works...
However, it is strnge that the icons don't appear in the Apps panel...

Thanks..
(I'm off until monday... )
Frilous

**Comment by kentonv** [comment]
If apps hang, it indicates a problem with the wildcard host. Are you sure `*.emhcalcul` has been configured as a wildcard in your DNS server?

See also: https://docs.sandstorm.io/en/latest/administering/install-troubleshooting/

**Comment by kentonv** [comment]
Hmm, I see you say it pinged OK. But the problem you're seeing almost always indicates that the wildcard isn't working right, and the message on the admin page indicates this too. Can you verify that the wildcard DNS actually points to the same IP address? I.e. `ping emhcalcul` and `ping foo.emhcalcul` should show the same destination address.

**Comment by frilous** [comment]
Hello,
Thank you for your answers.
I think I know why it doesn't work.
When I'm working on the server where sandstorm is installed, he is his own dns server, so he recognizes the wildcards. But when I am connected to another computer, he is looking for the dns server of the whole domain instead of searching on the local server. That could explain why it only works on the local server.
I don't think the It tema will let me use wildcards? Is it possible to do without wildcards ?

Thanks again...

Frilous

**Comment by ocdtrekkie** [comment]
@frilous It's not possible to use Sandstorm without wildcards. There's actually an article in the docs about why it is needed: https://docs.sandstorm.io/en/latest/administering/wildcard/
